apollo/resolvers/form/createForm.ts

```diff
@@ -0,0 +1,25 @@
+import { GQLContext } from '@/apollo/interfaces';
+import { Form } from '@/models/form';
+import { ApolloError } from 'apollo-server-micro';
+
+interface Args {
+  workspaceId: string;
+}
+
+export const createForm = async (
+  _: Record<string, never>,
+  { workspaceId }: Args,
+  { db, session }: GQLContext,
+) => {
+  const workspace = await db.workspaceRepo.get(workspaceId);
+
+  if (!workspace?.users.map(String).includes(session.user.id)) {
+    throw new ApolloError('Workspace not found', '404');
+  }
+
+  const form = new Form();
```

Comment:
Since this is t doing anything other than calling ‘new’ I think the instant ion of the form class should happen inside workspace and instead just call createForm()

Reply:
Thinking about this more I’m not sure it matters

Reply:
Do we want to keep the repos simple in a since to where they always take in data and do something with it? I guess that's kind of the way I saw it, but I could go either way. However, I think the creation of the form variable is an extra line I don't need here 😄 Just do `workspace.createForm(new Form())`

Reply:
First off this is technically an entity not a repo. The repo is just the data access wrapper abstraction. Thinking about it more it does matter that the form is created from outside the entity. I think the way the events line up there has to be an input param, otherwise it won’t know the uuid to store it with (the forms I’d field) in a situation like this. I like your reduction to one line. Moving forward (and keeping the existing behavior) any method on an entity should take params so that event timelines are consistent. (Or in the case of workspace init creating the id as a parameter default is fine too)

Reply:
Me typing fast at lunch the start of this sounds like it has 🔥- not my intention! Just trying to address the comment fully sorry for the wording